### PR TITLE
docs/os/mounting-storage: mount unit file location (Supersede #620)

### PR DESCRIPTION
**Pull request by joshix**
-Rewrite mounting-storage guide to incorporate the mention of unit file location without repeating it in each section
-Summarize the mount unit concept first, then get specific
-Pending review by @kayrus and @robszumski, this PR supersedes #620

### CODE REVIEW

File: os/mounting-storage.md [hunk 1]
```diff
@@ -101,7 +99,7 @@ Note the declaration of `ConditionPathExists=!/var/lib/docker.btrfs`. Without th
 
 ## Mounting NFS Exports
 
-This cloud-config excerpt enables the NFS host monitor [`rpc.statd(8)`](http://linux.die.net/man/8/rpc.statd), then mounts an NFS export onto the CoreOS system's `/var/www`. Mount unit names match the target mount point, with interior slashes replaced by dashes. A unit mounting onto `/var/www` is thus named `var-www.mount`.
+This cloud-config excerpt enables the NFS host monitor [`rpc.statd(8)`](http://linux.die.net/man/8/rpc.statd), then mounts an NFS export onto the CoreOS node's `/var/www`.
```
Comment (robszumski) [inline os/mounting-storage.md]: I've always tried to use "machine" here, but now that k8s uses "node" terminology (and we want to match upstream terms) this kind of breaks down. We should figure out a policy.

File: os/mounting-storage.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
+The [cloud-config]({{site.baseurl}}/docs/cluster-management/setup/cloudinit-cloud-config) *mount unit* mechanism is used to attach additional filesystems to CoreOS nodes, whether such storage is provided by an underlying cloud platform, physical disk, SAN, or NAS system. By [systemd convention](http://www.freedesktop.org/software/systemd/man/systemd.mount.html), mount unit names derive from the target mount point, with interior slashes replaced by dashes, and the `.mount` extension appended. A unit mounting onto `/var/www` is thus named `var-www.mount`.
+
+Mount units name the source filesystem and target mount point, and optionally the filesystem type. Cloud-config writes mount unit files beneath `/etc/systemd/system`. *Systemd* mounts filesystems defined in such units at boot time. The following example mounts an [EC2 ephemeral disk]({{site.baseurl}}/docs/running-coreos/cloud-providers/ec2/#instance-storage) at the node's `/media/ephemeral` directory, and is therefore named `media-ephemeral.mount`:
```
Comment (robszumski) [inline os/mounting-storage.md]: same node/machine issue here

**Comment by robszumski** [comment]
LGTM other than minor comments

**Comment by joshix** [comment]
I like node -- despite "virtual *machine*", that threadbare term that truly
describes every common timeshared, memory-virtualized process that has
executed since ~ the third research edition.

Also adopts upstream vocabulary.

We can reserve "machine" for future use, perhaps specific references to
discrete hardware, while anyone still remembers what that is.

Objections to this entry number one in the CoreOS style guide?

(This is worth asking marketing and/or polvi. It plants the first flag
in our evolving a stable definition of what "platform" now means.)

-J

**Pull request merged**